File: app/src/main/java/app/we/go/emojidraw/widget/CenteredToast.java
package app.we.go.emojidraw.widget;

import android.content.Context;
import android.support.annotation.StringRes;
import android.view.Gravity;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;


// Toast with centered text
public class CenteredToast {


    public static Toast makeText(Context context, @StringRes int message, int duration) {
        final Toast toast = Toast.makeText(context, message, duration);
        LinearLayout layout = (LinearLayout) toast.getView();
        if (layout.getChildCount() > 0) {
            TextView tv = (TextView) layout.getChildAt(0);
            tv.setGravity(Gravity.CENTER_HORIZONTAL);
        }
        return toast;
    }
}

File: app/src/main/java/app/we/go/emojidraw/widget/CenteredToast.kt
package app.we.go.emojidraw.widget

import android.content.Context
import androidx.annotation.StringRes
import android.view.Gravity
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast

// Toast with centered text
// TODO convert to an extension function on Activity?
object CenteredToast {

    fun show(context: Context, @StringRes message: Int, duration: Int) {
        val toast = Toast.makeText(context, message, duration)
        val layout = toast.view as LinearLayout
        (layout.getChildAt(0) as TextView?)?.apply {
            gravity = Gravity.CENTER_HORIZONTAL
        }
        toast.show()
    }
}
